decidim-budgets/app/forms/decidim/budgets/admin/order_reminder_form.rb

```diff
@@ -0,0 +1,42 @@
+# frozen_string_literal: true
+
+module Decidim
+  module Budgets
+    module Admin
+      class OrderReminderForm < Decidim::Form
+        def reminder_amount
+          @reminder_amount ||= begin
+            return 0 unless voting_enabled?
```

Comment:
In a perfect world, reminders would be generated in the morning and voting would stop in the evening. But yes with the current implementation reminder could be sent too "late". However, I do not know how the guard should be implemented, where we would get necessary information? For components under processes we could check if current step is ending soon and next step has voting disabled, but that's not possible with components under assemblies for example.

If we implement this, what would be time (before voting closes) for latest reminders?

Reply:
1. Taking into account that there could be delays in the multiple system involved (like the job queue, email delivery, email reception) after talking with @decidim/product we think that 6 hours before closing the voting would be great for leaving time for email reception and action. 
2. This would only happen in the cases where it can be inferred, even if it's a shortcut/happy path (like "6 hours before the step finishes if the component is related to a step"). In the cases where it can't be known when the voting finishes (like assemblies) then we will not do anything.

Reply:
This has been implemented now!